pkg/scheduler/server/server.go

```diff
@@ -267,3 +290,97 @@ func clientEndpoints(initialPeersListIP []string, idToPort map[string]string) []
 	}
 	return clientEndpoints
 }
+
+// runJobWatcher (dynamically) watches for (client) sidecar connections and adds them to the connection pool.
+func (s *Server) runJobWatcher(ctx context.Context) error {
+	log.Infof("Starting job watcher")
+
+	errCh := make(chan error)
+
+	s.jobWatcherWG.Add(2)
+
+	// Goroutine for handling sidecar connections
+	go func() {
+		defer log.Info("Sidecar connections goroutine shutting down.")
+		defer s.jobWatcherWG.Done()
+		s.handleSidecarConnections(ctx, errCh)
+	}()
+
+	// Goroutine for handling job streaming at trigger time
+	go func() {
+		defer log.Info("Job streaming goroutine shutting down.")
+		defer s.jobWatcherWG.Done()
+		s.handleJobStreaming(ctx, errCh)
+	}()
+
+	// Wait for any errors from either goroutine
+	select {
+	case err := <-errCh:
+		return err
+	case <-ctx.Done():
+		s.jobWatcherWG.Wait()
+		log.Info("JobWatcher go routines exited successfully")
+		return ctx.Err()
+	}
+}
+
+// handleJobStreaming handles the streaming of jobs to Dapr sidecars.
+func (s *Server) handleJobStreaming(ctx context.Context, errCh chan<- error) {
+	for {
+		select {
+		case job := <-s.jobTriggerChan:
+			log.Infof("Got the job at trigger time in the jobWatcher. Job: %+v", job)
+			metadata := job.GetMetadata()
+			appID := metadata["appID"]
+
+			jobUpdate := &schedulerv1pb.StreamJobResponse{
+				Data:     job.GetJob().GetData(),
+				Metadata: metadata,
+			}
+
+			namespace := job.GetNamespace()
+			// Pick a stream corresponding to the appID
+			stream, _, err := s.connectionPool.GetStreamAndContextForNSAppID(namespace + appID)
+			if err != nil {
+				log.Errorf("Error getting stream for appID: %v", err)
+				errCh <- err
+				continue
+			}
+			// Send the job update to the sidecar
+			if err := stream.Send(jobUpdate); err != nil {
+				log.Errorf("Error sending job at trigger time: %v", err)
```

Comment:
```suggestion
```

Reply:
same: I can change it to debug. I thought it was valuable to log.

Reply:
rn there isn't the orphan job go routine, so it would silently fail, so at the bare min I think we should log.

Reply:
I set this to debug